plugins/modules/yarn.py

```diff
@@ -244,8 +244,9 @@ def list(self):
         # because it only only lists binaries, but `yarn global add` can install libraries too.
         result, error = self._exec(cmd, run_in_check_mode=True, check_rc=False, unsupported_with_global=True)
 
-        if error:
-            self.module.fail_json(msg=error)
+        for line in error.splitlines():
+            if json.loads(line)['type'] == 'error':
```

Comment:
Is this always, ever going to be a list of lines, where every line is a JSON object? If not, it might make sense to wrap this in a `try/except` to fail with a nicer message (instead of a module crash).

Reply:
(And maybe it makes sense to move this into a separate function, so this code isn't repeated all over the place :) )

Reply:
I've added a function to handle this in https://github.com/ansible-collections/community.general/pull/6129/commits/5411c7aef9613ceb1be6ce387d57f0d20d979d07